projects/fusion-e2e/entities/tests/chart/chartLabelIcon.pw.ts

```diff
@@ -0,0 +1,50 @@
+import {expect, Locator, test} from '@playwright/test';
+import {ChartPage} from '../../pages/chart-page';
+import {chartLabelWithIconStoryId} from './consts';
+import {CART_LABELS_ICONS_MOCK} from '@ironsource/fusion-ui/components/chart-labels/v4/chart-labels-v4.stories.mock';
+import {toRgba} from '../utils';
+
+const ALPHA_VALUE = 100;
+
+let chartPage: ChartPage;
+
+test.beforeEach(async ({page}) => {
+    chartPage = new ChartPage(page);
+    await chartPage.goto({storyId: chartLabelWithIconStoryId});
+    await chartPage.waitForComponent();
+});
+
+test('Reloads the icon label component', async () => {
+    await chartPage.waitForIconLabelComponent();
+});
+
+test('Checks the number and properties of labels', async () => {
+    const labels: Locator = await chartPage.getLabels();
+    const labelsCount = await labels.count();
+    expect(labelsCount).toBe(CART_LABELS_ICONS_MOCK.length);
+
+    for (let i = 0; i < labelsCount; i++) {
+        await checkLabelProperties(i, labels);
+    }
+});
+
+async function checkLabelProperties(index: number, labels: Locator) {
```

Comment:
General note that should be applied for all tests - functions should be implemented under test util file instead of being directly in the tests file

Reply:
Done